x-pack/platform/plugins/shared/onechat/server/services/tools/builtin/definitions/create_visualization/guess_chart_type.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ModelProvider } from '@kbn/onechat-server';
+
+export async function getChartType(
+  modelProvider: ModelProvider,
+  existingType: string,
+  nlQuery: string
+) {
+  const model = await modelProvider.getDefaultModel();
+  const chartTypeResponse = await model.chatModel.invoke([
+    {
+      role: 'system',
+      content: `You are a data visualization expert. Based on the user's query, suggest the most appropriate chart type from the following options: metric, gauge, tagcloud, pie.
+
+Respond with ONLY the chart type name, nothing else.
```

Comment:
Same here, I would use structured output, not only it's simpler and avoids mistakes (smaller models may ignore instructions such as `and nothing else` and answer with a full sentence), and also it allows to add a `reasoning` or `explanation` parameter (which should be defined before the actual answering parameters) to help with the selection